DevCycle/SSEConnection.swift

```diff
@@ -11,21 +11,51 @@ import LDSwiftEventSource
 
 public typealias MessageHandler = (String) -> Void
 
-class SSEConnection {
-    private var connection: EventSource
+protocol SSEConnectionProtocol {
+    var connected: Bool { get }
+    func openConnection()
+    func close()
+    func reopen()
+}
+
+class SSEConnection: SSEConnectionProtocol {
+    private var connection: EventSource?
+    private var url: URL
+    private var handler: MessageHandler
+    
+    var connected: Bool
     
     init(url: URL, eventHandler: @escaping MessageHandler) {
+        self.url = url
+        self.handler = eventHandler
+        self.connected = false
+        self.openConnection()
+    }
+    
+    public func openConnection() {
         Log.debug("Establishing realtime streaming connection.")
-        let handler = Handler(handler: eventHandler)
+        let handler = Handler(handler: self.handler)
         self.connection = EventSource(config: EventSource.Config(
             handler: handler,
             url: url
         ))
-        self.connection.start()
+        self.connection?.start()
```

Comment:
any way to do this refactoring without making this optional?

Reply:
swift's requirement for properties that are non-optional have to be initialized before any methods called on the instance, unfortunately. It guarantees it won't be optional, so unless i also initialize `self.connection` in the `init`, this is about as clean as i can get it

Reply:
is there anything wrong with just setting the connection up in init? Also it can't figure out that you're calling the method that sets it up from `init`?

Reply:
yeah i can set it up in init, but i'd be duplicating code. And yes it explicitly disallows me from calling any instance methods before the properties that are not optional are initialized with a value